Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/NAPPALifecycleObserver.java

```diff
@@ -0,0 +1,30 @@
+package nl.vu.cs.s2group.nappa;
+
+import android.app.Activity;
+import android.util.Log;
+
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleObserver;
+import androidx.lifecycle.OnLifecycleEvent;
+
+public class NAPPALifecycleObserver implements LifecycleObserver {
```

Comment:
This PR feature would require overriding the `onPause` method in addition to the `onResume` method. 
After reading the Android lifecycle and doing some tests, I verified that it would be best to not reuse the `onResume` to store the visit time since `onPause` is always called but the next `onResume` might not be called (e.g. closing the app)

While reading the Android lifecycle I came across the `LifecycleObserver` class, which allows us to define what to do in which cycle (e.g. `onResume`, `onCreate`, `onPause`, etc.) and the activities of the app can simply add this observer.